src/components/branding/che-branding.ts

```diff
@@ -50,12 +54,19 @@ export class CheBranding {
    * Update branding data.
    */
   private updateData(): Promise<void> {
-    return angular.element.get(`${ASSET_PREFIX}product.json`).then(branding => {
+    return angular.element.getJSON(`${ASSET_PREFIX}product.json`).then(branding => {
```

Comment:
It does not have a sense after the gulp task 'branding'.

Reply:
With these changes, we generate a file with constant during each build.
Should we request the same content in additional?

Reply:
Not sure I've got what you mean, could you elaborate more on this?

Reply:
We should decide what to use. We can generate and use the file with constants or we should request a 'product.json' file. And we have a modal error message without branding.

I prefer the next idea - all branding resources move to a separate container.

Reply:
Correct me if I'm wrong, but in our previous discussions we've decided:

1. get rid of branding data duplication in the dashboard project
2. `product.json` should contain only patch to overwrite default branding.
3. show an error page in case if `product.json` is not available (doesn't matter empty or not)

And there was nothing about not to request the `product.json`.

Please read also the description to [this issue](https://github.com/eclipse/che/issues/17741).

Reply:
I confirm - we're going to merge two sources - one from source-code + product.json that is going to be mandatory requested, and is supposed to contain patch from downstream. Dashboard is failed to load if product.json is not available.